components/TARGET_PSA/services/storage/common/psa_storage_common_impl.h

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2019 ARM Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __PSA_STORAGE_COMMON_IMPL_H__
+#define __PSA_STORAGE_COMMON_IMPL_H__
+
+#include "psa/error.h"
+#include "psa/storage_common.h"
+#include "KVStore.h"
+
+#ifdef   __cplusplus
+extern "C"
+{
+#endif
+
+typedef struct {
+    uint32_t major;
+    uint32_t minor;
+} psa_storage_version_t;
+
+typedef psa_status_t (*migrate_func_t)(mbed::KVStore *kvstore, const psa_storage_version_t *old_version, const psa_storage_version_t *new_version);
+
+void psa_storage_handle_version(mbed::KVStore *kvstore, const char *version_key, const psa_storage_version_t *version,
```

Comment:
no documentation for these functions? _impl.h are internal headers?

Reply:
Yes, they are. They are the worker functions for the PS & ITS features.